src/components/Tokens/TokenDetails/PriceChart.tsx

```diff
@@ -190,25 +199,30 @@ export function PriceChart({ width, height }: PriceChartProps) {
         pricePoint = x0.valueOf() - d0.timestamp.valueOf() > d1.timestamp.valueOf() - x0.valueOf() ? d1 : d0
       }
 
-      setSelected({ pricePoint, xCoordinate: timeScale(pricePoint.timestamp) })
+      setCrosshair(timeScale(pricePoint.timestamp))
+      setDisplayPrice(pricePoint)
     },
     [timeScale, pricePoints]
   )
 
+  if (!hasData) {
+    return null
+  }
+
```

Comment:
is `null` definitely what you want to return when there's no data? I know there's a loading state for the price chart--we should probably also have a "no data" or error state instead of not rendering the element at all. (mb if this already exists)